tests/Doctrine/Tests/Common/Annotations/AnnotationRegistryTest.php

```diff
@@ -168,4 +168,25 @@ public function testResetClearsRegisteredAutoloaderFailures() : void
 
         self::assertSame(2, $failures);
     }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testRegisterLoaderIfNotExistsOnlyRegisteresSameLoaderOnce() : void
+    {
+        $className = 'autoloadedClass' . random_int(10, 100000);
+        AnnotationRegistry::reset();
+        $i = 0;
+        $autoLoader = function () use (&$i) : bool {
+            $i += 1;
+            return false;
+        };
+        AnnotationRegistry::registerUniqueLoader($autoLoader);
+        AnnotationRegistry::registerUniqueLoader($autoLoader);
+        AnnotationRegistry::registerUniqueLoader($autoLoader);
```

Comment:
Is this test about multiple registrations of the same loader or about the loader having returned `false` shouldn't be called again? If it's about the former, then there should be two calls to `AnnotationRegistry::registerUniqueLoader()` and one to  `AnnotationRegistry::loadAnnotationClass()`, not three of each.

Reply:
return false is just because the loader itself does not define the class (and the AnnotationRegistry expects the loader to return false when it does not make the class exist after the call)

Reply:
The return value of `false` is fine. I'm asking for a minimal sufficient set of method calls. The fact that `AnnotationRegistry::registerUniqueLoader()` is called more than twice and `AnnotationRegistry::loadAnnotationClass()` is called more than once seems redundant. Isn't it?